lib/fluent/plugin/out_s3.rb

```diff
@@ -147,8 +146,8 @@ def ensure_bucket
     end
 
     def check_apikeys
-      @bucket.empty?
```

Comment:
This probably will affect some users who have minimalistic IAM policies. Furthermore, using `exists?` does not check for access permissions, so I think it defeats the purpose of `check_apikeys`.

See the discussion here: https://github.com/fluent/fluent-plugin-s3/issues/80